### PR TITLE
[Serve] Rename Serve in Documentation

### PR DESCRIPTION
**Pull request by bllchmbrs**
## Why are these changes needed?

As we discussed, we want to migrate from RayServe to Ray Serve.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

**Review by deanwampler — APPROVED** [review]
+1

**Comment by simon-mo** [comment]
You also need to change `index.rst` TOC and reference

**Review by simon-mo — APPROVED** [review]

**Comment by anabranch** [comment]
Think we are good since I used `:ref` call.

**Comment by anabranch** [comment]
NVM, think I found what you were referring to.

File: doc/source/index.rst [hunk 1]
```diff
@@ -280,7 +280,7 @@ Getting Involved
 
 .. toctree::
    :maxdepth: -1
-   :caption: RayServe
+   :caption: Ray Serve
 
    rayserve/overview.rst
```
Comment (simon-mo) [inline doc/source/index.rst:285]: These need to be updated to `serve/...`

Comment (anabranch) [reply doc/source/index.rst]: I think I got them when you said this, they're correct in the current PR right?

**Comment by simon-mo** [comment]
/home/travis/build/ray-project/ray/doc/source/serve/overview.rst:4:Title underline too short.

```
Ray Serve: Scalable and Programmable Serving
===========================================
```

**Pull request merged**